src/ascii/port.rs

```diff
@@ -924,7 +1009,26 @@ impl<'a, B: Backend> Port<'a, B> {
                 let mut response = self.build_response()?;
                 response = header_check(&response).check(response)?;
                 let response = R::try_from(response).map_err(AsciiUnexpectedResponseError::new)?;
```

Comment:
Since checks now always work on `AnyResponse`, should we do this afterwards to avoid unnecessary conversions?

Reply:
The response is now converted late (but the conversion error is still checked early